processing/src/main/java/io/druid/query/QueryToolChest.java

```diff
@@ -186,13 +205,83 @@
    * be queried.  It can use whatever criteria it wants in order to do the pruning, it just needs to
    * return the list of Segments it actually wants to see queried.
    *
-   * @param query The query being processed
+   * @param query    The query being processed
    * @param segments The list of candidate segments to be queried
-   * @param <T> A Generic parameter because Java is cool
+   * @param <T>      A Generic parameter because Java is cool
+   *
    * @return The list of segments to actually query
    */
   public <T extends LogicalSegment> List<T> filterSegments(QueryType query, List<T> segments)
   {
     return segments;
   }
+
+  /**
+   * @param filter Filter input to optimize
+   *
+   * @return Returns an optimized {@link DimFilter}
+   * Optimization is done by rewriting all ExtractionDimFilter of type {@link LookupExtractionFn} as a clause of literals {@link SelectorDimFilter}
+   */
+  public static DimFilter optimizeExtractionDimFilter(final DimFilter filter)
```

Comment:
this method feels out of place in the toolchest

is there not a common file related to filters we can use?